auth/options.go

```diff
@@ -93,9 +95,44 @@ func getRequestDetails(ctx context.Context) (string, string, error) {
 	return path.Dir(fullMethodString)[1:], path.Base(fullMethodString), nil
 }
 
+// WithTLS gathers metadata from a TLS-authenticated client
+func WithTLS() option {
+	f := func(ctx context.Context) ([]*pdp.Attribute, error) {
+		p, ok := peer.FromContext(ctx)
+		if !ok || p.AuthInfo == nil {
+			return []*pdp.Attribute{
+				{Id: tlsVerified, Type: "boolean", Value: "false"},
+			}, nil
+		}
+
+		t, ok := p.AuthInfo.(*credentials.TLSInfo)
+		if !ok || t.State.VerifiedChains == nil || len(t.State.VerifiedChains) == 0 || len(t.State.VerifiedChains[0]) == 0 {
```

Comment:
Can `t.State` ever be `nil` here? If so, there should be a `nil` check on `t.State` before referencing the `VerifiedChains` field.

Reply:
t.State is a struct